app/models/upload.rb

```diff
@@ -112,8 +112,17 @@ def purge_storage
       chunk.purge_storage
       chunk.destroy
     end
-    storage_provider.delete_object_manifest(storage_container, id)
-    self.update(purged_on: DateTime.now)
+
+    begin
+      storage_provider.delete_object_manifest(storage_container, id)
+      self.update(purged_on: DateTime.now)
+    rescue StorageProviderException => e
+      if e.message.match /Not Found/
+        self.update(purged_on: DateTime.now)
```

Comment:
Could you put the `self.update` line after the begin/rescue block, instead of repeating it twice?